ansible_runner/display_callback/module.py

```diff
@@ -21,18 +21,35 @@
 import collections
 import contextlib
 import datetime
+import os
 import sys
 import uuid
 from copy import copy
 
 # Ansible
 from ansible import constants as C
 from ansible.plugins.callback import CallbackBase
-from ansible.plugins.callback.default import CallbackModule as DefaultCallbackModule
+from ansible.plugins.loader import callback_loader
 
-# AWX Display Callback
 from .events import event_context
-from .minimal import CallbackModule as MinimalCallbackModule
+
+user_specified_callback = os.getenv('ORIGINAL_STDOUT_CALLBACK')
+runner_callback = os.environ.pop('ANSIBLE_STDOUT_CALLBACK', None)
+
+if user_specified_callback:
+    os.environ['ANSIBLE_STDOUT_CALLBACK'] = user_specified_callback
+
+is_adhoc = os.getenv('AD_HOC_COMMAND_ID', False)
+
+# Dynamically construct base classes for our callback module, to support custom stdout callbacks.
+if is_adhoc:
+    default_stdout_callback = 'minimal'
+else:
+    default_stdout_callback = C.config.get_config_value('DEFAULT_STDOUT_CALLBACK')
+
+DefaultCallbackModule = callback_loader.get(default_stdout_callback).__class__
```

Comment:
Huh, so compared to #387, it looks like the key difference is that you don't try to change the "awx_display" name. You swap out the super class instead. This may resolve some of my hesitations about this.

In unpredictable contexts, I got errors importing and using the Ansible plugin loaders inside of other plugins. Since this runs on the control machine, it might be less likely to be a problem.